### PR TITLE
WIP: add Windows support

### PR DESCRIPTION
**Pull request by gillins**
Attempt to add Windows support. 

Seems the problem with Python 3.5 builds was that uicc.exe [does not run on Windows 7](https://connect.microsoft.com/VisualStudio/feedback/details/806268/uicc-exe-of-windows-kits-8-1-can-not-run-on-windows-7-sp1-x86).

I have worked around this by removing all the Windows only GUI stuff. I don't *think* that we need this.

### CODE REVIEW

**Comment by gillins** [comment]
I think this is good to merge. Everything builds except the `leash` Windows GUI which I have disabled. This is because:

1. It uses MFC which does not come with Visual Studio Express prior to 2015
2. On VS 2015, the uicc.exe utility which is needed for compiling the "ribbon" GUI does not run on Windows 7 which appears to be what appveyor uses. I see the same thing on my PC

I had to do a bit of tweaking to get it going with VS2008 (for Python 2.7) so this will be a bit of a maintenance issue in future, same with many packages these days I suppose.

**Comment by jakirkham** [comment]
@patricksnape or @msarahan, could one of you please review this? 😄

**Comment by jakirkham** [comment]
Thanks for doing this @gillins.

File: recipe/run_test.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+if sys.platform == 'win32':
+    libfreexl = ctypes.CDLL('krb5_%s.dll' % bits)
+elif sys.platform == 'darwin':
+    # LD_LIBRARY_PATH not set on OSX or Linux
+    path = os.path.expandvars('$PREFIX/lib/libkrb5.dylib')
+    libfreexl = ctypes.CDLL(path)
+else:
+    path = os.path.expandvars('$PREFIX/lib/libkrb5.so')
+    libfreexl = ctypes.CDLL(path)
+
```
Comment (patricksnape) [inline recipe/run_test.py]: Missing new line

**Comment by patricksnape** [comment]
Heroic effort! All looks good to me.

**Comment by ocefpaf** [comment]
Merging this without Travis-CI as the changes, except for the new tests\*, only affects Windows.

\*And we know from past builds that the new tests are passing.

Thanks @gillins! Another great contribution to the Windows user community!!

**Pull request merged**